### PR TITLE
Fix default argument value type checking

### PR DESCRIPTION
**Pull request by NotLebedev**
# Description
Fix type checking in arguments default values not adhering to subtyping rules
Currently following examples produce a parse error:
```nu
def test [ --qwe: record<a: int> = {a: 1 b: 1} ] { }
def test [ --qwe: list<any> = [ 1 2 3 ] ] { }
```
despite types matching. Type equality check is replaced with subtyping check and everything parses fine:
# User-Facing Changes
Default values of flag arguments type checking behavior is in line with `let` statements

### CODE REVIEW

**Review by amtoine — CHANGES_REQUESTED** [review]
@NotLebedev 
i'm not 100% sure i see what this changes, could you think of some tests to add to make sure this works in the future? :yum: 
that will also help see what had to be fixed :relieved:

File: tests/parsing/mod.rs [hunk 1]
```diff
@@ -244,3 +245,16 @@ fn parse_long_duration() {
 
     assert_eq!(actual.out, "1min 18sec 797ms");
 }
+
+
+#[rstest]
+#[case("def test [ --a: any = 32 ] {}")]
```
Comment (amtoine) [inline tests/parsing/mod.rs:251]: ooh ok, good :+1: 

i see this first case is the only one to work now, all the other ones appear to fail :ok_hand: 
let's see if the CI goes green :relieved:

**Comment by NotLebedev** [comment]
This changes how type of `--` flags were checked against their default value. Previously following things (I added them to tests) were invalid
image
because parser checked for equality of types, not compatibility with regards to  subtyping rules. This is especially problematic because there was no way to set a default value if type was complex and had `any`. This PR changes type checking to be in line with `let` definitions (uses exact same function to check if types are compatible)

**Comment by amtoine** [comment]
appreciate the work @NotLebedev :pray: 

with my knowledge of the type system, this looks good to me, especially with the tests that were not working and do work now :ok_hand: 

can you fix the merge conflicts? :innocent:

**Comment by amtoine** [comment]
thanks for fixing the conflicts, let's land this and see how it goes :pray: :partying_face:

**Pull request merged**